v2/tracing/tracing.go

```diff
@@ -5,57 +5,80 @@ import (
 
 	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus"
 	"github.com/Azure/go-shuttle/v2"
-	"github.com/devigned/tab"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	traceCarrierField = "traceCarrier"
+	serviceTracerName = "go-shuttle"
 )
 
+var tracer trace.Tracer
+
+func init() {
+	tp := tracesdk.NewTracerProvider(tracesdk.WithSampler(tracesdk.AlwaysSample()))
```

Comment:
we are in a library here. shouldn't this come from the current settings of the service that uses this library?

Reply:
are there any examples on how libraries allow to integrate with services that uses opentelemetry for tracing?